src/tests/ftest/cart_fault_injection/pool_fault_inject.py

```diff
@@ -0,0 +1,225 @@
+        self.look_missed_request(self.pool.dmg.result.stderr)
+
+    def _clean(self):
+        """Destroy container and pool
+        Look for missed request on each executed command
+        """
+        self.container.destroy()
+        self.look_missed_request(
+            self.container.daos.result.stderr)
+        self.pool.destroy(force=1)
+        self.look_missed_request(self.pool.dmg.result.stderr)
+
+    def test_pool_services(self):
+        """ Test the following pool commands:
+                Create
+                Delete
+                Create container
+                Write container (not a pool command)
+                Exclude rank from pool
+                Reintegrate rank to pool
+                List
+                Destroy
+
+            All the request have fault injection of a probability of 5% which
+            attempts to mimic a flaky internet connection. Daos tools should
+            re-attempt the requests without failing.
+
+            The only passes if the following two conditions pass:
+                All commands executed successfully
+                There was at least one missed request, which should be an
+                injected fault of real network issue.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,large
+        :avocado: tags=pool_with_faults,test_pool_services
+        """
+        failed_commands = 0
+        dmg_command = self.get_dmg_command()
+        pool_namespace = self.params.get("pool", "/run/*")
+        number_pools = self.params.get("number_pools", "/run/*")
+
+        for _ in range(number_pools):
+            try:
+                # Create pool
+                self.pool = self.get_pool(namespace=pool_namespace)
+                self.look_missed_request(self.pool.dmg.result.stderr)
+
+                # Container section
+                self.create_and_write_into_container()
+
+                # Remove rank and add it back
+                self.exclude_and_reintegrate()
+
+                # Simple dmg commands
+                dmg_command.system_query()
+                self.look_missed_request(dmg_command.result.stderr)
+                dmg_command.pool_list()
+                self.look_missed_request(dmg_command.result.stderr)
+
+                # Delete pool and container
+                self._clean()
+
+            except CommandFailure as e:
+                self.log.error(str(e))
+                failed_commands += 1
+            except DaosTestError as e:
+                self.log.error(str(e))
+                failed_commands += 1
+            finally:
+                if self.pool:
+                    self._clean()
+        end_message = f"Failed commands: {failed_commands} \n" \
+                      f"Failed requests: {self.failed_requests}"
+        self.log.info(end_message)
+        assert failed_commands == 0, "Execution of command(s) failed."
+        assert self.failed_requests > 0, "No faults detected."
```

Comment:
```suggestion
        self.assertEqual(failed_commands, 0, "Execution of command(s) failed.")
        self.assertGreater(self.failed_requests, 0, "No faults detected.")
```
Is there any reason you're not using the self.assertEqual/Greater? That's what most of the tests are using.

Reply:
For no particular reason, I can change it to use those functions (y)

Reply:
This should do it.